juniper_official/Solutions/Traffic-Blackhole/trap-stats.rule

```diff
@@ -0,0 +1,115 @@
+/*
+	* REQ: COL-1109
+	* Rule to get Trap Statistics (Drop Counters) for Blackhole Detection.
+	* Each platform will have different trap names.
+*/
+healthbot {
+	topic hardware.fpc {
+		rule trap-stats {
+			keys [ fpc-name trap-name ];
+			synopsis "Forwarding Block Trap Names and Values";
+			description "Forwarding Block Trap Names and Values";
+			sensor sensor-trap-stats {
+				synopsis "Forwarding Block Trap Names and Values";
+				description "Forwarding Block Trap Names and Values";
+				open-config {
+					sensor-name /junos/system/linecard/packet/usage/;
+					frequency 60s;
+				}
+			}
+			field fpc-name {
+				sensor sensor-trap-stats {
+					where "/components/component/@name =~ /PIC[0-9]:NPU/";
+					where "/components/component/properties/property/@name != 'ts-input-packets-pps'";
+					where "/components/component/properties/property/@name != 'ts-output-packets-pps'";
+					path "/components/component/@name";
+				}
+				type string;
+				description "FPC name";
+			}
+			field trap-name {
+				sensor sensor-trap-stats {
+					where "/components/component/@name =~ /PIC[0-9]:NPU/";
+					where "/components/component/properties/property/@name != 'ts-input-packets-pps'";
+					where "/components/component/properties/property/@name != 'ts-output-packets-pps'";
+					path "/components/component/properties/property/@name";
+				}
+				type string;
+				description "Trap Name";
+			}
+			field trap-value {
+				sensor sensor-trap-stats {
+					where "/components/component/@name =~ /PIC[0-9]:NPU/";
+					where "/components/component/properties/property/@name != 'ts-input-packets-pps'";
```

Comment:
Remove this from here. There is no need to duplicate the where clause

Reply:
Removed the where clauses from the field sensors. Only kept the where clauses in the key (fpc-name)